cmd/config-sync/config_sync.go

```diff
@@ -171,3 +202,257 @@ func (c *ConfigSync) syncConfig(desired *qdr.BridgeConfig) error {
+			return configSync.ensureSslProfile(added.SslProfile, added.SslProfile, sharedTlsFilesDir)
+		}
+	}
+
+	for _, deleted := range changes.HttpListeners.Deleted {
+		if len(deleted.SslProfile) > 0 {
+			log.Printf("Deleting cert files related to HTTP Listener sslProfile %s", deleted.SslProfile)
+
+			agent, err := configSync.agentPool.Get()
+			if err != nil {
+				return err
+			}
+
+			if err = agent.Delete("io.skupper.router.sslProfile", deleted.SslProfile); err != nil {
+				return fmt.Errorf("Error deleting ssl profile: #{err}")
+			}
+
+			err = os.RemoveAll(sharedTlsFilesDir + "/" + deleted.SslProfile)
+			if err != nil {
+				return err
+			}
+
+		}
+
+	}
+
+	for _, deleted := range changes.HttpConnectors.Deleted {
+		if len(deleted.SslProfile) > 0 {
+			log.Printf("Deleting cert files related to HTTP Connector sslProfile %s", deleted.SslProfile)
+			err := os.RemoveAll(sharedTlsFilesDir + "/" + deleted.SslProfile)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (c *ConfigSync) checkSecrets(desired *qdr.BridgeConfig, sharedTlsFilesDir string) error {
+
+	for _, listener := range desired.HttpListeners {
+		if len(listener.SslProfile) > 0 {
+			err := c.ensureSslProfile(listener.SslProfile, listener.SslProfile, sharedTlsFilesDir)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	for _, connector := range desired.HttpConnectors {
+		if len(connector.SslProfile) > 0 {
+			err := c.ensureSslProfile(connector.SslProfile, connector.SslProfile, sharedTlsFilesDir)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (c *ConfigSync) checkConnectorSecrets(desired map[string]qdr.ConnectorStatus, sharedTlsFilesDir string) error {
+
+	agent, err := c.agentPool.Get()
+	if err != nil {
+		return err
+	}
+
+	for _, connectorStatus := range desired {
+		connector, err := agent.GetConnectorByName(connectorStatus.Name)
+		if err != nil {
+			return err
+		}
+
+		if len(connector.SslProfile) > 0 {
+			secretName := strings.TrimSuffix(connector.SslProfile, "-profile")
+			err := c.ensureSslProfile(connector.SslProfile, secretName, sharedTlsFilesDir)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (c *ConfigSync) syncConnectorSecrets(changes *qdr.ConnectorDifference, sharedTlsFilesDir string) error {
+
+	agent, err := c.agentPool.Get()
+	if err != nil {
+		return err
+	}
+
+	for _, added := range changes.Added {
+		if len(added.SslProfile) > 0 {
+			sslProfile := changes.AddedSslProfiles[added.SslProfile]
+			log.Printf("Creating ssl profile %s", sslProfile.Name)
+			err := agent.CreateSslProfile(sslProfile)
+			if err != nil {
+				return err
+			}
+
+			log.Printf("Copying cert files related to Connector sslProfile %s", added.SslProfile)
+			secretName := strings.TrimSuffix(added.SslProfile, "-profile")
+
+			return c.ensureSslProfile(added.SslProfile, secretName, sharedTlsFilesDir)
```

Comment:
It should only return on error. Otherwise the deleted entries won't be processed.

Reply:
Fixed! thanks!